spec/taskBehaviors.js

```diff
@@ -205,6 +206,74 @@ describe('Tasks', function() {
             expect(runValues).toEqual([1,3]);
         });
     });
+
+    describe('runEarly', function() {
+        it('Should run tasks early', function() {
+            var runValues = [];
+            var func = function(value) {
+                runValues.push(value);
+            };
+            ko.tasks.schedule(func.bind(null, 1));
+            expect(runValues).toEqual([]);
+
+            // Immediately runs any scheduled tasks
+            ko.tasks.runEarly();
+            expect(runValues).toEqual([1]);
+
+            // Skip calling jasmine.Clock.tick to show that the queue is clear
+        });
+
+        it('Should run tasks early during task processing', function() {
+            var runValues = [];
+            var func = function(value) {
+                runValues.push(value);
+            };
+
+            // Schedule two tasks; the first one schedules other tasks and calls runEarly
+            ko.tasks.schedule(function() {
+                ko.tasks.schedule(func.bind(null, 2));
+                expect(runValues).toEqual([]);
+
+                ko.tasks.runEarly();
+                expect(runValues).toEqual([1,2]);
+
+                // Schedule another task; it will be run after this one completes
+                ko.tasks.schedule(func.bind(null, 3));
+            });
+            ko.tasks.schedule(func.bind(null, 1));
+
+            jasmine.Clock.tick(1);
+            expect(runValues).toEqual([1,2,3]);
+        });
+
+        it('Should keep correct state if a task throws an exception', function() {
+            var runValues = [];
+            var func = function(value) {
+                runValues.push(value);
+            };
+            ko.tasks.schedule(func.bind(null, 1));
+            ko.tasks.schedule(function() {
+                expect(runValues).toEqual([1]);
+                ko.tasks.runEarly();
+                ko.tasks.schedule(func.bind(null, 3));      // This never gets called
```

Comment:
As per my comment above, I would find it less surprising and more useful if this *did* get called.